terraform/workspaces/infra/network/logs.tf

```diff
@@ -1,17 +1,73 @@
 resource "aws_flow_log" "main" {
-  iam_role_arn    = aws_iam_role.vpc_flow_logs.arn
-  log_destination = aws_cloudwatch_log_group.main.arn
-  traffic_type    = "ALL"
-  vpc_id          = aws_vpc.app.id
+  log_destination_type = "s3"
+  log_destination      = aws_s3_bucket.flow_logs.arn
+  traffic_type         = "ALL"
+  vpc_id               = aws_vpc.app.id
 
   tags = {
     Name = "${var.workspace}-vpc-logs"
   }
 }
 
+resource "aws_s3_bucket" "flow_logs" {
+  bucket = "${var.workspace}-vpc-logs"
+}
+
+resource "aws_s3_bucket_versioning" "flow_logs" {
+  bucket = aws_s3_bucket.flow_logs.id
+
+  versioning_configuration {
+    status = "Enabled"
```

Comment:
What's the value of having versioning enabled?

Reply:
It is recommended from a security perspective since it would be more difficult for an attacker to cover their tracks. In practice every object should only have one version.